### PR TITLE
Add Account structure

### PR DESCRIPTION
**Pull request by sheapwilliams**
- Ensure cache line alignment

### CODE REVIEW

File: include/monad/core/account.hpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+#include <monad/config.hpp>
+#include <monad/core/bytes.hpp>
+#include <monad/core/int.hpp>
+
+MONAD_NAMESPACE_BEGIN
+
+struct Account {
+    uint256_t balance{0};
+    bytes32_t code_hash{NULL_HASH};
```
Comment (jhunsaker) [inline include/monad/core/account.hpp:12]: let's use uint256 for hashes and values - migration will be using bit_cast in the meantime

better alignment and seems more conceptually correct to me (like hash values used as indexes)

File: include/monad/core/account.hpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    uint64_t nonce{0};
+    uint64_t incarnation{0};
+};
+
+MONAD_NAMESPACE_END
```
Comment (jhunsaker) [inline include/monad/core/account.hpp:20]: add sizeof and alignment assertions

File: include/monad/core/bytes.hpp [hunk 3]
```diff
@@ -11,4 +11,6 @@ using bytes32_t = ::evmc::bytes32;
 static_assert(sizeof(bytes32_t) == 32);
 static_assert(alignof(bytes32_t) == 1);
 
+inline constexpr bytes32_t NULL_HASH{0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470_bytes32};
```
Comment (sheapwilliams) [inline include/monad/core/bytes.hpp:15]: Still need to change this type.

Comment (sheapwilliams) [reply include/monad/core/bytes.hpp:15]: Ok, bytes32_t is a big endian type.  uint256_t is a native type.  I'll revert this for now.

**Pull request merged**